cartographer/mapping/internal/2d/scan_matching/real_time_correlative_scan_matcher_2d.cc

```diff
@@ -108,29 +148,28 @@ double RealTimeCorrelativeScanMatcher2D::Match(
 }
 
 void RealTimeCorrelativeScanMatcher2D::ScoreCandidates(
-    const ProbabilityGrid& probability_grid,
-    const std::vector<DiscreteScan2D>& discrete_scans,
+    const Grid2D& grid, const std::vector<DiscreteScan2D>& discrete_scans,
     const SearchParameters& search_parameters,
     std::vector<Candidate2D>* const candidates) const {
   for (Candidate2D& candidate : *candidates) {
-    candidate.score = 0.f;
-    for (const Eigen::Array2i& xy_index :
-         discrete_scans[candidate.scan_index]) {
-      const Eigen::Array2i proposed_xy_index(
-          xy_index.x() + candidate.x_index_offset,
-          xy_index.y() + candidate.y_index_offset);
-      const float probability =
-          probability_grid.GetProbability(proposed_xy_index);
-      candidate.score += probability;
+    if (grid.GetGridType() == GridType::PROBABILITY_GRID) {
```

Comment:
use switch case (with break;) without default instead. That would fail compiling if someone adds a new grid type.

Reply:
Ok, sounds good. I'll update the CeresScanMatcher as well.